examples/ember-cli/todomvc/app/components/todo-item.js

```diff
@@ -0,0 +1,49 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+	repo: Ember.inject.service(),
+	tagName: 'li',
+	editing: false,
+	classNameBindings: ['todo.completed', 'editing'],
+
+	actions: {
+		startEditing() {
+			this.get('onStartEdit')();
+			this.set('editing', true);
+			Ember.run.scheduleOnce('afterRender', this, 'focusInput');
+		},
+
+		doneEditing(e) {
+			let newTitle = e.target.value.trim();
```

Comment:
minor opportunity for destructuring param. or use `value="target.value"` in the template?